lib/ansible/plugins/terminal/nxos.py

```
@@ -48,6 +50,47 @@ class TerminalModule(TerminalBase):
         re.compile(br"invalid (.+?)at '\^' marker", re.I)
     ]
 
+    def on_become(self, passwd=None):
+        if self._get_prompt().endswith(b'enable#'):
+            return
+
+        out = self._exec_cli_command(b'show privilege')
+        out = to_text(out, errors='surrogate_then_replace').strip()
+        if 'Disabled' in out:
+            raise AnsibleConnectionFailure('Feature privilege is not enabled')
+
+        # if already at privilege level 15 return
+        if '15' in out:
+            return
+
+        cmd = {u'command': u'enable'}
+        if passwd:
+            cmd[u'prompt'] = to_text(r"(?i)[\r\n]?Password: $", errors='surrogate_or_strict')
+            cmd[u'answer'] = passwd
+            cmd[u'prompt_retry_check'] = True
+
+        try:
+            self._exec_cli_command(to_bytes(json.dumps(cmd), errors='surrogate_or_strict'))
+            prompt = self._get_prompt()
+            if prompt is None or not prompt.strip().endswith(b'enable#'):
+                raise AnsibleConnectionFailure('failed to elevate privilege to enable mode still at prompt [%s]' % prompt)
+        except AnsibleConnectionFailure as e:
+            prompt = self._get_prompt()
+            raise AnsibleConnectionFailure('unable to elevate privilege to enable mode, at prompt [%s] with error: %s' % (prompt, e.message))
+
+    def on_unbecome(self):
+        prompt = self._get_prompt()
+        if prompt is None:
+            # if prompt is None most likely the terminal is hung up at a prompt
+            return
```

Should we raise a exception here rather silent return ?